app/src/main/java/org/wikipedia/feed/announcement/Announcement.kt

```diff
@@ -0,0 +1,116 @@
+package org.wikipedia.feed.announcement
+
+import androidx.annotation.NonNull
+import androidx.annotation.Nullable
+import com.google.gson.annotations.SerializedName
+import org.wikipedia.json.annotations.Required
+import org.wikipedia.util.DateUtil
+import java.util.*
+
+class Announcement {
+
+    private var action: Action? = null
+
+    @SerializedName("negative_text")
+    var negativeText: String? = null
+
+    @Required
+    var id: String = ""
+
+    @SerializedName("start_time")
+    @Required
+    private val startTime: String? = null
+
+    @SerializedName("end_time")
+    @Required
+    private val endTime: String? = null
+    private val border: Boolean? = null
+
+    @Required
+    var text: String? = null
+
+    @Required
+    val type: String = ""
+
+    @SerializedName("image_url")
+    @Nullable
+    var imageUrl: String? = null
+
+    @SerializedName("caption_HTML")
+    val footerCaption: String = ""
+
+    @SerializedName("image_height")
+    val imageHeight: String = ""
+
+    @SerializedName("logged_in")
+    val loggedIn: Boolean? = null
+
+    @SerializedName("reading_list_sync_enabled")
+    val readingListSyncEnabled: Boolean? = null
+
+    @SerializedName("min_version")
+    val minVersion: String? = null
+
+    @SerializedName("max_version")
+    val maxVersion: String? = null
+
+    val beta: Boolean? = null
+    val platforms = emptyList<String>()
+    val countries = emptyList<String>()
+    val placement: String = PLACEMENT_FEED
+
+    constructor() {}
+
+    constructor(@NonNull id: String?,
+                @NonNull text: String?,
+                @NonNull imageUrl: String?,
+                @NonNull action: Action?,
+                @NonNull negativeText: String?) {
+        this.id = id!!
+        this.text = text!!
+        this.imageUrl = imageUrl!!
```

Comment:
Not sure but maybe adding `.orEmpty()`, or change the variables to a non-null type?